pyccel/codegen/printing/cwrappercode.py

```diff
@@ -464,6 +524,197 @@ def get_PyBuildValue(self, used_names, variable):
+            mini_wrapper_func_body += [FunctionCall(Py_DECREF, [i]) for i in self._to_free_PyObject_list]
+            mini_wrapper_func_body.append(Return(wrapper_results))
+            self._to_free_PyObject_list.clear()
+            # Building Mini wrapper function
+            mini_wrapper_func_name = self.get_new_name(used_names.union(self._global_names), func.name.name + '_mini_wrapper')
+            self._global_names.add(mini_wrapper_func_name)
+
+            mini_wrapper_func_def = FunctionDef(name = mini_wrapper_func_name,
+                arguments = parse_args,
+                results = wrapper_results,
+                body = mini_wrapper_func_body,
+                local_vars = mini_wrapper_func_vars.values())
+            funcs_def.append(mini_wrapper_func_def)
+
+            # append check condition to the functioncall
+            body_tmp.append((PyccelEq(check_var, LiteralInteger(flags)), [AliasAssign(wrapper_results[0],
+                    FunctionCall(mini_wrapper_func_def, parse_args))]))
+
+        # Errors / Types management
+        # Creating check_type function
+        check_func_def = self._create_wrapper_check(check_var, parse_args, types_dict, used_names, funcs[0].name.name)
+        funcs_def.append(check_func_def)
+
+        # Create the wrapper body with collected informations
+        body_tmp = [((PyccelNot(check_var), [Return([Nil()])]))] + body_tmp
+        body_tmp.append((LiteralTrue(),
+            [PyErr_SetString('PyExc_TypeError', '"Arguments combinations don\'t exist"'),
+            Return([Nil()])]))
+        wrapper_body_translations = [If(*body_tmp)]
+
+        # Parsing Arguments
+        parse_node = PyArg_ParseTupleNode(python_func_args, python_func_kwargs, funcs[0].arguments, parse_args, keyword_list, True)
+        wrapper_body += list(default_value.values())
+        wrapper_body.append(If((PyccelNot(parse_node), [Return([Nil()])])))
+
+        #finishing the wrapper body
+        wrapper_body.append(Assign(check_var, FunctionCall(check_func_def, parse_args)))
+        wrapper_body.extend(wrapper_body_translations)
+        wrapper_body.append(Return(wrapper_results)) # Return
+
+        # Create FunctionDef
+        funcs_def.append(FunctionDef(name = wrapper_name,
+            arguments = wrapper_args,
+            results = wrapper_results,
+            body = wrapper_body,
+            local_vars = wrapper_vars.values()))
+
+        sep = self._print(SeparatorComment(40))
+
+        return sep + '\n'.join(CCodePrinter._print_FunctionDef(self, f) for f in funcs_def)
+
+    def _create_wrapper_check(self, check_var, parse_args, types_dict, used_names, func_name):
+        check_func_body = []
+        flags = (len(types_dict) - 1) * 4
+        for a in types_dict:
+            var_name = ""
+            body = []
+            types = []
+            l = list(types_dict[a])
+            l.sort(key= lambda x : x[0].precision)
+            for s in l:
```

Comment:
Can you use more descriptive names please. `a`, `l` and `s` is too many one letter variables for me to remember what is what